File: mobile-apps/alloy-android/app/src/main/kotlin/com/alloyremote/alloy/model/ConnectionState.kt
package com.alloyremote.alloy.model

sealed class ConnectionState {
    data object Disconnected : ConnectionState()
    data object Connecting : ConnectionState()
    data object Connected : ConnectionState()
    data class Failed(val message: String) : ConnectionState()

    val userFacingLabel: String
        get() = when (this) {
            Disconnected -> "Disconnected"
            Connecting -> "Connecting…"
            Connected -> "Connected"
            is Failed -> "Error: $message"
        }
}

File: mobile-apps/alloy-ios/Alloy/Model/ConnectionState.swift
import Foundation

enum ConnectionState: Equatable, Sendable {
    case disconnected
    case connecting
    case connected
    case failed(String)

    var isConnected: Bool {
        if case .connected = self { return true }
        return false
    }

    var userFacingLabel: String {
        switch self {
        case .disconnected: return "Disconnected"
        case .connecting:   return "Connecting…"
        case .connected:    return "Connected"
        case .failed(let msg): return "Error: \(msg)"
        }
    }
}
